massa-network/src/peer_info_database.rs

```diff
@@ -823,19 +738,25 @@ impl PeerInfoDatabase {
                 ));
             }
         }
+        let peer = if let Some(p) = self.get_peer_type(ip) {
+            p
+        } else {
+            let p = PeerInfo::new(*ip, false);
+            self.peers.insert(*ip, p);
+            self.get_peer_type(ip).unwrap()
+        };
+
+        if self.is_max_in_connection_count_reached(peer) {
```

Comment:
this should be checked first, otherwise we might have already inserted a new peer (3 lines above)

Reply:
Why is it wrong ?

Reply:
because if config.max_in_connections = 0 and if the peer didn't exist in self.peers before,  this function will fail because max peers connections is reached, but the peer will remain in self.peers (without pruning nor anything).

The cleanest way is always to check if we can create/insert something before doing so

Reply:
> but the peer will remain in self.peers (without pruning nor anything).

Why it's bad ?

Reply:
always check if we can insert something before actually inserting it